altapay.php

```diff
@@ -0,0 +1,2353 @@
+        $helper->orderWay = 'ASC';
+        $content = Terminal::getTerminals();
+
+        return $helper->generateList($content, $fields_list);
+    }
+
+    /**
+     * Validate merchant details form
+     */
+    private function postValidation()
+    {
+        if (Tools::isSubmit('btnSubmit')) {
+            if (!Tools::getValue('ALTAPAY_USERNAME')) {
+                $this->postErrors[] = $this->l('API username is required');
+            } elseif (!Tools::getValue('ALTAPAY_URL')) {
+                $this->postErrors[] = $this->l('API URL is required');
+            }
+            if (!filter_var(Tools::getValue('ALTAPAY_URL'), FILTER_VALIDATE_URL)) {
+                $this->postErrors[] = $this->l('Incorrect format for the API URL - Use "https://paymentURL"');
+            }
+        }
+    }
+
+    /**
+     * Method for saving gateway configuration details in plugin settings
+     */
+    private function postProcess()
+    {
+        if (Tools::isSubmit('btnSubmit')) {
+            Configuration::updateValue('ALTAPAY_USERNAME', Tools::getValue('ALTAPAY_USERNAME'));
+            $urlPath = preg_replace('/\s+/', '', Tools::getValue('ALTAPAY_URL'));
+            if (Tools::substr($urlPath, -1) != '/') {
+                Configuration::updateValue('ALTAPAY_URL', $urlPath .= '/');
+            } elseif (Tools::substr($urlPath, -1) == '/') {
+                Configuration::updateValue('ALTAPAY_URL', $urlPath);
+            }
+            if (Tools::getValue('ALTAPAY_PASSWORD') !== "") {
+                Configuration::updateValue('ALTAPAY_PASSWORD', Tools::getValue('ALTAPAY_PASSWORD'));
+            }
+            if (Tools::getValue('AUTOCAPTURE_STATUSES') !== "") {
+                Configuration::updateValue('AUTOCAPTURE_STATUSES', serialize(Tools::getValue("AUTOCAPTURE_STATUSES")));
+            }
+        }
+        $this->Mhtml .= '<div class="alert alert-success"> ' . $this->l('Settings updated') . '</div>';
+    }
+
+    /**
+     * Displays the error that occurred in the hookActionOrderStatusUpdate method, if any.
+     *
+     * @param $params
+     * @return mixed
+     */
+    public function hookBackOfficeHeader($params)
+    {
+        $cookie = $this->context->cookie;
+
+        $this->context->controller->addJquery();
+        $this->context->controller->addJS($this->_path . 'views/js/form.js', 'all');
+
+        if ($cookie->altapayError) {
+            $this->context->controller->errors[] = $cookie->altapayError;
+
+            // unset the variable:
+            $cookie->altapayError = null;
+        }
+
+        // always returns false because there is no template to display
+        return false;
+    }
+
+    /**
+     * Captures a payment when the status is changed to Shipped
+     *
+     * @param $params
+     * @return mixed
+     */
+    public function hookActionOrderStatusUpdate($params)
+    {
+        $results = $this->selectOrder($params);
+
+        if (!$results) {
+            return;
```

Comment:
you cannot mix void returns with other types, please use a explicit null return:
```suggestion
            return null;
```